systemd/emulation.go

```diff
@@ -153,17 +155,28 @@ func (s *emulation) AddMountUnitFile(snapName, revision, what, where, fstype str
 		return "", fmt.Errorf("cannot mount %s (%s) at %s in preseed mode: %s; %s", what, hostFsType, where, err, string(out))
 	}
 
-	multiUserTargetWantsDir := filepath.Join(dirs.SnapServicesDir, "multi-user.target.wants")
-	if err := os.MkdirAll(multiUserTargetWantsDir, 0755); err != nil {
-		return "", err
+	// In order to handle preseeding of old images with old version of snapd,
+	// we also need to add the unit to multi-user.target
+	targets := []string{
+		"snapd.mounts.target",
+		"multi-user.target",
```

Comment:
It might not be obvious why we use `snapd.mounts.targets`, since the comment mentions only `multi-user.target`. Maybe add a comment

    // snapd.mounts is the primary target for mounting the snaps.

before the "In order to..." one.

Reply:
Done